plugins/lookup/passwordstore.py

```diff
@@ -582,9 +582,9 @@ def run(self, terms, variables, **kwargs):
                     if self.paramvals['overwrite']:
                         with self.opt_lock('write'):
                             result.append(self.update_password())
-                    elif self.paramvals["subkey"] != "password" and not self.passdict.get(self.paramvals['subkey']):  # password exists but not the subkey
-                        with self.opt_lock('write'):
-                            result.append(self.update_password())
+                        if self.paramvals["subkey"] != "password" and not self.passdict.get(self.paramvals['subkey']):  # password exists but not the subkey
+                            with self.opt_lock('write'):
+                                result.append(self.update_password())
```

Comment:
I'm not sure whether this is correct. Right before the `if`, you're already running `result.append(self.update_password())`, and now, under a certain condition, you run it *a second time*.